### DISCUSSION TITLE
Generate String() methods for model types

### DISCUSSION THREAD

**Discussion by leventov** [99designs/gqlgen#1935] [Ideas]
Would be nice if gqlgen generated `String()` methods for model types. GraphQL's `Boolean`, `Float`, `Int` translate into Go's `*bool`, `*float64`, `*int` which are not printed well by Go's `"%#v"` formatting.

**Comment by frederikhors** [comment]
Can you write here an example, please? We can help then with PR.

**Comment by leventov** [comment]
GraphQL type:
```
type MyType {
  flag: Boolean
}
```

Generated Go code:
```
type MyType struct {
	flag *bool
}

func (t *MyType) String() string {
	flagS := "null"
	if (t.flag != nil) {
		flagS := strconv.FormatBool(*t.flag)
	}
	return fmt.Sprintf("MyType{flag: %s}", flagS)
}
```

**Comment by frederikhors** [comment]
Where are they using “%#v”?

Are they using this for bool for example? https://github.com/99designs/gqlgen/blob/master/graphql/bool.go

**Comment by leventov** [comment]
Who are "they"?

We are using %#v because this is the best we can have (for debug logging, for example) in the absence of generated `String()` method, but it doesn't print the values of all the optional fields (and optional sub-structures) because %#v doesn't dereference pointers in Go.
